src/features/offer/helpers/getOfferTags/getOfferTags.ts

```diff
@@ -0,0 +1,22 @@
+import { OfferExtraData } from 'api/gen'
+
+export function getOfferTags(subcategoryLabel: string, extraData?: OfferExtraData) {
+  const tags: string[] = [subcategoryLabel]
+
+  const pushTag = (mainType: string, subType?: string | null) => {
+    tags.push(mainType)
+    if (subType) {
+      tags.push(subType)
+    }
+  }
+
+  if (extraData?.musicType) {
+    pushTag(extraData.musicType, extraData?.musicSubType)
+  }
+
+  if (extraData?.showType) {
+    pushTag(extraData.showType, extraData?.showSubType)
+  }
+
+  return tags
+}
```

Comment:
```suggestion

export function getOfferTags(subcategoryLabel: string, extraData?: OfferExtraData) {
  const tags: string[] = [subcategoryLabel]

  const maybePushTag = (tag: string | null | undefined) => {
    if (tag) tags.push(tag)
  }

  maybePushTag(extraData?.musicType)
  maybePushTag(extraData?.musicSubType)
  maybePushTag(extraData?.showType)
  maybePushTag(extraData?.showSubType)

  return tags
}
```

Je te propose ça : plus simple, plus flat, et pas besoin d'autant de conditions.

En poussant un peu, on peut même faire mieux

```suggestion
export function getOfferTags(subcategoryLabel: string, extraData?: OfferExtraData) {
  return [
    subcategoryLabel,
    extraData?.musicType,
    extraData?.musicSubType,
    extraData?.showType,
    extraData?.showSubType,
  ].filter(Boolean)
}
```

Et là tu enlèves toute la complexité inutile.

Reply:
J'aime bien la 2ème proposition

Reply:
Il va peut-être falloir que tu overrides le typage qui ne va pas comprendre et dire `<string | undefined>[]`